records/generator.go

```diff
@@ -231,14 +252,66 @@ func sanitizedSlaveAddress(hostname string, spec labels.HostNameSpec) string {
 	return address
 }
 
+func containerIPFromDockerInspect(inspect string) (string, error) {
+	var inspectJSON []struct {
+		NetworkSettings struct {
+			IpAddress string `json:"IpAddress"`
+		} `json:"NetworkSettings"`
+	}
+
+	err := json.Unmarshal([]byte(inspect), &inspectJSON)
```

Comment:
`inspectJSON` is a reference type so no need to pass a reference in to `json.Unmarshal`